elastic/apm-server
# Remove `http.request.socket` fields

## Motivation/summary

Remove some unused, non-indexed fields, to tidy up our data model and bring it more in line with ECS.

Stop recording `http.request.socket.encrypted`, with no replacement. This field is not used in the UI, and is redundant; the URL scheme can be used instead.

Stop recording `http.request.socket.remote_address`. Instead, record the direct network peer IP and port in `source.ip` and `source.port`.

## Checklist

~- [ ] Documentation has been updated~

## How to test these changes

1. Run `apm-server -E instrumentation.enabled=true`
2. `curl -H "X-Forwarded-For: 1.2.3.4" http://localhost:8200/foo`
3. Wait for a transaction to be indexed
4. Check that the transaction does _not_ have any `http.request.socket` fields, but has `source.ip` (127.0.0.1) and `source.port` set (client's bind port, pass `--local-port` to curl to control this) and `client.ip` (1.2.3.4)

## Related issues

https://github.com/elastic/apm-server/issues/4410
https://github.com/elastic/apm-server/issues/4120
